core-blocks/latest-comments/edit.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	PanelBody,
+	RangeControl,
+	ToggleControl,
+	ServerSideRender,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import {
+	InspectorControls,
+	BlockAlignmentToolbar,
+	BlockControls,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies.
+ */
+import './editor.scss';
+
+const MIN_COMMENTS = 1;
+const MAX_COMMENTS = 100;
+
+class LatestComments extends Component {
+	constructor() {
+		super( ...arguments );
+		this.toggleHandler = this.toggleHandler.bind( this );
+		this.changeCommentsToShow = this.changeCommentsToShow.bind( this );
+	}
+
+	toggleHandler( propName ) {
+		return () => {
```

Comment:
Ideally, this arrow function should be promoted to a class method and bind to this instead of `toggleHandler` to avoid unnecessary rerenders.

Reply:
In this case the whole element re-renders from server-side data though, right? 🤷‍♂️

Reply:
Wait, scratch that, I see. I think for now it's okay as we use this pattern elsewhere in the codebase, but if it's a performance issue we could see about fixing it everywhere. I'm afraid I've recommended this in the past, not realising the potential for unneeded re-renders...

Reply:
Yes, I saw it in other places, too. I don't think this is a big issue for leaf components. However, it's something we should try to avoid whenever it doesn't cause too much work.